scripts/install_dev_osgeo.sh

```diff
@@ -0,0 +1,177 @@
+
+CRAN=${CRAN_SOURCE:-"https://cloud.r-project.org"}
+echo "options(repos = c(CRAN = '${CRAN}'))" >>"${R_HOME}/etc/Rprofile.site"
+
+# a function to install apt packages only if they are not installed
+function apt_install() {
+    if ! dpkg -s "$@" >/dev/null 2>&1; then
+        if [ "$(find /var/lib/apt/lists/* | wc -l)" = "0" ]; then
+            apt-get update
+        fi
+        apt-get install -y --no-install-recommends "$@"
+    fi
+}
+
+function url_latest_gh_released_asset() {
+    wget -qO- "https://api.github.com/repos/$1/releases/latest" | grep -oP "(?<=\"browser_download_url\":\s\")https.*\.tar.gz(?=\")"
+}
+
+export DEBIAN_FRONTEND=noninteractive
+apt_install \
+    gdb \
+    git \
+    libcairo2-dev \
+    libcurl4-openssl-dev \
+    libexpat1-dev \
+    libpq-dev \
+    libsqlite3-dev \
+    libudunits2-dev \
+    make \
+    pandoc \
+    qpdf \
+    sqlite3 \
+    subversion \
+    valgrind \
+    vim \
+    tk-dev \
+    wget \
+    libv8-dev \
+    libjq-dev \
+    libprotobuf-dev \
+    libxml2-dev \
+    libprotobuf-dev \
+    protobuf-compiler \
+    unixodbc-dev \
+    libssh2-1-dev \
+    libgit2-dev \
+    libnetcdf-dev \
+    locales \
+    libssl-dev \
+    libtiff-dev \
+    cmake \
+    libtiff5-dev
+
+LD_LIBRARY_PATH=/usr/local/lib:$LD_LIBRARY_PATH
+
+# install proj
+# https://download.osgeo.org/proj/
+if [ "$PROJ_VERSION" = "latest" ]; then
+    PROJ_DL_URL=$(url_latest_gh_released_asset "OSGeo/PROJ")
+else
+    PROJ_DL_URL="http://download.osgeo.org/proj/proj-${PROJ_VERSION}.tar.gz"
+fi
+
+wget "$PROJ_DL_URL" -O proj.tar.gz
+tar zxvf proj.tar.gz
+rm proj.tar.gz
+cd proj-*
+mkdir build
+cd build
+cmake ..
+make
+make install
+cd ../..
+ldconfig
+
+# install gdal
+# https://download.osgeo.org/gdal/
+if [ "$GDAL_VERSION" = "latest" ]; then
+    GDAL_DL_URL=$(url_latest_gh_released_asset "OSGeo/gdal")
+else
+    GDAL_DL_URL="https://download.osgeo.org/gdal/${GDAL_VERSION}/gdal-${GDAL_VERSION}.tar.gz"
+fi
+
+wget "$GDAL_DL_URL" -O gdal.tar.gz
+tar -xf gdal.tar.gz
+rm gdal.tar.gz
+cd gdal*
+./configure
+make
+make install
+cd ..
+ldconfig
+
+# install geos
+# https://libgeos.org/usage/download/
+if [ "$GEOS_VERSION" = "latest" ]; then
+    GEOS_VERSION=$(wget -qO- "https://api.github.com/repos/libgeos/geos/git/refs/tags" | grep -oP "(?<=\"ref\":\s\"refs/tags/)\d+\.\d+\.\d+" | tail -n -1)
+fi
+
+wget http://download.osgeo.org/geos/geos-"${GEOS_VERSION}".tar.bz2
+bzip2 -d geos-*bz2
+tar xf geos*tar
+rm geos*tar
+cd geos*
+./configure
+make
+make install
+cd ..
+ldconfig
+
+install2.r --error --skipmissing --skipinstalled -n "$NCPUS" \
+    sp \
+    rgeos \
+    rgdal \
+    RPostgreSQL \
+    RSQLite \
+    testthat \
+    knitr \
+    tidyr \
+    geosphere \
+    maptools \
+    maps \
+    microbenchmark \
+    raster \
+    dplyr \
+    tibble \
+    units \
+    DBI \
+    covr \
+    protolite \
+    tmap \
+    mapview \
+    odbc \
+    pool \
+    rmarkdown \
+    RPostgres \
+    spatstat \
+    stars
+
```

Comment:
this list is just copied from install_geospatial.sh package list, and may not be in sync with it.  Some of these are slated for deprecation (rgdal, rgeos) or succeeded by other packages, and a good number don't bind the geospatial C libs so could be installed from binary (though some of those will pull in packages that do, so might need testing.  

I don't think we need to change this now, but just something for us to think about going forward -- could get some input from the spatial devs and revise this list with some of the newer state-of-the-art at the same time.

Reply:
@cboettig 

At this time, the R packages installed by the two scripts appear to be quite different. The following packages are only installed by dev_osgeo.

```
covr, DBI, dplyr, knitr, maps, microbenchmark, odbc, pool, protolite, rmarkdown, RPostgres, RPostgreSQL, RSQLite, testthat, tibble, tidyr, units
```

The following packages are installed only in geospatial.

```
BiocManager, classInt, deldir, geoR, gstat, hdf5r, lidR, mapdata, ncdf4, proj4, RandomFields, RColorBrewer, rlas, RNetCDF, spacetime, spatialreg, spdep, terra, tidync
```

I would suggest removing packages that exist only in dev_osgeo.
Most of these seem to be essentially irrelevant to dev_osgeo.